### PR TITLE
Clients - React Native - Upgrade RN to 51 - Remove webpack config not needed anymore

### LINKED ISSUE #289: Mobile - Move away from webpack-config
@expo/webpack-config is no longer receiving updates. So it is sort of a time bomb until this is going to be a blocker for using new features from expo.

https://docs.expo.dev/router/migrate/from-expo-webpack/

The only reason we're keeping this is `rn-bounceable` which is the only the only reference in our current webpack config file. I opened a ticket a while back in the repository but I don't think this library is going to receive more updates https://github.com/kanzitelli/rn-bounceable/issues/7#issue-1877324590

### PR DESCRIPTION
**Pull request by lakardion**
## What this does

Upgrades expo to [51](https://expo.dev/changelog/2024/05-07-sdk-51)

Closes #289

### CODE REVIEW

**Review by oudeismetis — DISMISSED** [review]
Looks good. Needs rebase.
Do we have a plan for fixing that mobile check that has been failing for a couple of weeks?

**Comment by lakardion** [comment]
> Looks good. Needs rebase. Do we have a plan for fixing that mobile check that has been failing for a couple of weeks?

Lemme see, looks like a 

> Looks good. Needs rebase. Do we have a plan for fixing that mobile check that has been failing for a couple of weeks?

Ah good call. There was a problem with the general gh actions file for linting mobile

**Comment by lakardion** [comment]
@oudeismetis I think the actions broke or I broke them. They're just stuck in Queued

File: {{cookiecutter.project_slug}}/clients/web/vue3/src/services/users/api.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ const resetPassword = createCustomServiceCall(
   },
   async ({ client, input, utils }) => {
     const { email, ...rest } = utils.toApi(input)
-    const res = await client.post(`/password/reset/code/confirm/${input.email}/`, rest)
+    const res = await client.post(`/password/reset/code/confirm/${email}/`, rest)
```
Comment (lakardion) [inline {{cookiecutter.project_slug}}/clients/web/vue3/src/services/users/api.js:32]: Fixes eslint issue

File: {{cookiecutter.project_slug}}/clients/web/vue3/src/views/ResetPassword.vue [hunk 2]
```diff
@@ -25,15 +25,14 @@
...
-
         <button class="btn--primary bg-primary" type="submit">Reset Password</button>
       </form>
     </div>
   </div>
 </template>
 
 <script>
-import { ref, watch } from 'vue'
+import { ref } from 'vue'
```
Comment (lakardion) [inline {{cookiecutter.project_slug}}/clients/web/vue3/src/views/ResetPassword.vue:35]: fixes eslint issue

**Comment by lakardion** [comment]
> Looks good. Needs rebase.
Do we have a plan for fixing that mobile check that has been failing for a couple of weeks?

Okay everything is passing now I'm going to merge considering the previous approval

**Pull request merged**